### ISSUE TITLE
Integration with ira/mackup broken

### ISSUE THREAD

**Issue by lyveng** [hluk/CopyQ#1196]
**Describe the bug**
[lra/mackup](https://github.com/lra/mackup) is a tool that is used for backing up application settings to cloud storage(dropbox, google drive, ..) which makes it easy to setup your dev environment when switching dev box. When we try to backup copyq we are facing lra/mackup#659

This is happening because the file ~/.config/copyq/.copyq_s is of type socket. Should this ideally be inside the config directory?

**To Reproduce**
1. [lra/mackup](https://github.com/lra/mackup) and copyq should be installed
2. copyq should be running
3. mackup config should not have disabled copyq
4. Run `mackup backup`
5. See error

**Expected behavior**
Not sure about it. Would be cool if .config/copyq had just the configuration and the socket file is present in /tmp. Also the backup works when copyq is not running since the socket file is created only when copyq is started. If you are fine with this behaviour, I can send a PR for this.

**Screenshots**
```bash
➜  ~ mackup backup
Backing up .config/copyq ...
Traceback (most recent call last):
  File "/home/livingstone.se/.local/bin/mackup", line 11, in <module>
    load_entry_point('mackup==0.8.26', 'console_scripts', 'mackup')()
  File "/home/livingstone.se/.local/lib/python3.7/site-packages/mackup/main.py", line 91, in main
    app.backup()
  File "/home/livingstone.se/.local/lib/python3.7/site-packages/mackup/application.py", line 107, in backup
    utils.copy(home_filepath, mackup_filepath)
  File "/home/livingstone.se/.local/lib/python3.7/site-packages/mackup/utils.py", line 102, in copy
    shutil.copytree(src, dst)
  File "/usr/lib/python3.7/shutil.py", line 365, in copytree
    raise Error(errors)
shutil.Error: [('/home/livingstone.se/.config/copyq/.copyq_s', '/home/livingstone.se/Dropbox/Mackup/.config/copyq/.copyq_s', "[Errno 6] No such device or address: '/home/livingstone.se/.config/copyq/.copyq_s'")]
➜  ~ ls -l .config/copyq/.copyq_s 
srwxr-xr-x 1 livingstone.se livingstone.se 0 Aug 15 04:09 .config/copyq/.copyq_s
```

**Version, OS and Environment**
(Get details from `copyq version` command if possible.)
 - Application Version - v3.8.0
 - OS - Debian Buster
 - Desktop environment, window manager - Wayland and Gnome

**Comment by hluk** [comment]
It's in configuration directory for safety reasons so that only single instance of the app can read/write settings and data.

The backup will be broken in some rare cases when CopyQ is writing configuration or data to the directory.

**Comment by lyveng** [comment]
> It's in configuration directory for safety reasons so that only single instance of the app can read/write settings and data.

Any reason putting the file in the system temp directory like `/tmp` would affect the safety aspect?

> The backup will be broken in some rare cases when CopyQ is writing configuration or data to the directory.

I'm assuming the directory you've referred to here is `~/.config/copyq`. It is not a rare case though. It always happens and it always fails when trying to copy the socket file.

**Comment by hluk** [comment]
> Any reason putting the file in the system temp directory like `/tmp` would affect the safety aspect?

See #330 and the fix in 1b25b20a3b216550cb08947ac6a372c091a9d4d1.

> It is not a rare case though. It always happens and it always fails when trying to copy the socket file.

What I meant is that if the back up starts while CopyQ writes configuration or data the backup can contain an empty configuration file, a missing data file or some additional temporary files (config and data are written to a temporary file before saving).

CopyQ should still recover if you restore such backup (without the socket file) but I couldn't find a safer location for the socket file. It's not very nice but having the socket file in configuration directory ensures that only one application instance can access the configuration and data.

**Comment by lyveng** [comment]
Thanks for pointing it out. Got your point. Since backup(using [mackup tool](https://github.com/lra/mackup)) is not a continuous process and is generally triggered manually, I think it is better to stop copyq before running the backup(also avoids the potentially empty config files issue)
